lib/bots/versionbot.ts

```diff
@@ -269,70 +334,86 @@ export class VersionBot extends GithubBot.GithubBot {
+            this.logger.log(LogLevel.INFO, "No valid 'Change-Type' tag found, failing last commit " +
                 `for ${owner}/${name}#${pr.number}`);
-            return this.gitCall(githubApi.repos.createStatus, {
-                        context: 'Versionist',
-                        description: 'None of the commits in the PR have a `Change-Type` tag',
-                        owner,
-                        repo: name,
-                        sha: head.sha,
-                        state: 'failure'
+            return this.githubCall({
+                data: {
+                    context: 'Versionist',
+                    description: 'None of the commits in the PR have a `Change-Type` tag',
+                    owner,
+                    repo: name,
+                    sha: head.sha,
+                    state: 'failure'
+                },
+                method: ghApiCalls.repos.createStatus,
             }).then(() => {
                 // We only complain on opening. Either at that point it'll be fixed, or we
                 // simply don't pass it in future. This stops the PR being spammed on status changes.
-                if (data.action === 'opened') {
-                    this.gitCall(githubApi.issues.createComment, {
-                        body: `@${data.sender.login}, please ensure that at least one commit contains a` +
-                            '`Change-Type:` tag.',
-                        owner,
-                        number: pr.number,
-                        repo: name,
+                if (event.cookedEvent.data.action === 'opened') {
+                    return this.githubCall({
+                        data: {
+                            body: `@${event.cookedEvent.data.sender.login}, please ensure that at least one commit ` +
+                                'contains a `Change-Type:` tag.',
+                            owner,
+                            number: pr.number,
+                            repo: name,
+                        },
+                        method: ghApiCalls.issues.createComment,
                     });
                 }
             });
         }).then(() => {
             // Get the labels for the PR.
-            return this.gitCall(this.githubApi.issues.getIssueLabels, {
-                number: pr.number,
-                owner,
-                repo: name
+            return this.githubCall({
+                data: {
+                    number: pr.number,
+                    owner,
+                    repo: name
+                },
+                method: ghApiCalls.issues.getIssueLabels
             });
-        }).then((labels: GithubBotApiTypes.IssueLabel[]) => {
+        }).then((labels: GithubApiTypes.IssueLabel[]) => {
             // If we don't have a relevant label for merging, we don't proceed.
             if (_.filter(labels, (label) => {
                 return label.name === MergeLabel;
             }).length !== 0) {
```

Comment:
I know it's not part of the PR but I'd use `_.any` for this, means that you don't have to loop through all if you don't need to.